### ISSUE TITLE
make comma optional

### ISSUE THREAD

**Issue by pannous** [henry-luo/mark#3]
`{input type:"email", id:"email"} ` ok
`{input type:"email" id:"email"} ` ok

**Comment by DanCech** [comment]
I was about to ask why the comma is needed at all

**Comment by henry-luo** [comment]
Thanks for the feedback.

Comma is added merely to make Mark a superset of JSON.  And I think JSON has comma added, because it needs to follow JS syntax.

Now since Mark has already broken syntax compatibility with JS, we may consider removing the comma between properties, and it will be more like HTML/XML with less delimiter noise.

I think I can consider making comma optional, as you suggested, just like ';' is optional in JS.

However, if we do that, people may start to ask making ',' optional in array. Should that be optional as well?

**Comment by pannous** [comment]
Good point. Thinking about it: Yes, commas should be optional everywhere:  
`{input type:"email" id:"email" class=[bold large] style={color:blue width:120} "enter@your.email.com"  }`

**Comment by pannous** [comment]
In one distant utopian future these variants will all have the same meaning
```
style:{color:blue width:120}
style={color:blue width=120}
style{color:blue width:120}
{style color:blue width:120}
```
people will cringe at remembering xml, where attributes and children where different.
